my-app/src/main/java/com/JavaWebServer/app/ClientSocket.java

```diff
@@ -0,0 +1,70 @@
+package com.JavaWebServer.app;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+
+public class  ClientSocket implements IClient {
+
+  private java.net.Socket socket;
+
+  public ClientSocket (java.net.Socket socket) {
+    this.socket = socket;
+  }
+
+  public Request getRequest() {
+    try{
+      BufferedReader request = new BufferedReader(getStreamReader()); 
+      return new Request(request.readLine());  
+    } catch (IOException e) {
+      System.out.println("cant read request"+ e);
+      return null;
```

Comment:
What happens if this returns `null`? It is used by `pathExists`, which doesn't seem to handle a `null` request.